pkg/registry/extensions/rest/storage_extensions.go

```diff
@@ -62,7 +81,13 @@ func (p RESTStorageProvider) v1beta1Storage(apiResourceConfigSource serverstorag
 	storage["daemonsets/status"] = daemonSetStatusStorage
 
 	//deployments
-	deploymentStorage := deploymentstore.NewStorage(restOptionsGetter)
+	appsdeploymentStorage := deploymentstore.NewStorage(restOptionsGetter)
+	deploymentStorage := DeploymentStorage{
+		Deployment: appsdeploymentStorage.Deployment,
+		Status:     appsdeploymentStorage.Status,
+		Scale:      appsdeploymentStorage.Scale,
+		Rollback:   &RollbackREST{*appsdeploymentStorage.Rollback},
```

Comment:
Instead of doing this, just pass this new thing where it's actually used, below.